server/integration/endpoint/src/main/java/org/infinispan/server/endpoint/deployments/AbstractCacheStoreExtensionProcessor.java

```diff
@@ -0,0 +1,52 @@
+package org.infinispan.server.endpoint.deployments;
+
+import org.infinispan.server.endpoint.Constants;
+import org.infinispan.server.endpoint.subsystem.ExtensionManagerService;
+import org.jboss.as.server.deployment.DeploymentPhaseContext;
+import org.jboss.msc.service.Service;
+import org.jboss.msc.service.ServiceBuilder;
+import org.jboss.msc.service.ServiceController;
+import org.jboss.msc.service.ServiceName;
+import org.jboss.msc.value.InjectedValue;
+
+import static org.infinispan.server.endpoint.EndpointLogger.ROOT_LOGGER;
+
+public abstract class AbstractCacheStoreExtensionProcessor<T> extends AbstractServerExtensionProcessor<T> {
+
```

Comment:
I have some doubts to this part of the implementation.... All deployed services implement Infinispan core interfaces (like `AdvancedCacheLoader` for example). Those classes are instantiated by `AbstractServerExtensionProcessor` in `addServices` method. This leads to Classloader issues when deployed JAR does not contain `Dependencies: org.infinispan services` in `MANIFEST.MF`. I'm not sure if this is correct. I have tried to invoke this fragment in overriden `deploy` method but it didn't solve this problem:
```java
private static final ModuleIdentifier API = ModuleIdentifier.create("org.infinispan");
...
@Override
public void deploy(DeploymentPhaseContext ctx) throws DeploymentUnitProcessingException {
    DeploymentUnit deploymentUnit = ctx.getDeploymentUnit();
    ModuleSpecification moduleSpec = deploymentUnit.getAttachment(Attachments.MODULE_SPECIFICATION);
    ModuleLoader moduleLoader = Module.getBootModuleLoader();
    moduleSpec.addSystemDependency(new ModuleDependency(moduleLoader, API, false, false, false, false));
}
```